.travis.yml

```diff
@@ -0,0 +1,55 @@
+---
+sudo: true
+
+addons:
+  apt:
+    packages:
+      - docker-ce
+      - python
+      - realpath
+      - ruby
+
+language: go
+
+env:
+  global:
+    - GO_METALINTER_VERSION="v2.0.11"
+
+go:
+  # Kubernetes minimum_go_version can be found in this file:
+  # https://github.com/kubernetes/kubernetes/blob/master/hack/lib/golang.sh
+  - master
+  - stable
+  - "1.10"
+  - "1.9"   # kubernetes 1.10 min was 1.9.1
+
+install:
+  - gem install asciidoctor mdl
+  - pip install --user --upgrade pip
+  - pip install --user yamllint
+  # install dep
+  - >
+    curl -L
+    https://raw.githubusercontent.com/golang/dep/master/install.sh
+    | sh
+  # install gometalinter
+  - >
+    curl -L
+    'https://raw.githubusercontent.com/alecthomas/gometalinter/master/scripts/install.sh'
+    | bash -s -- -b ${TRAVIS_HOME}/gopath/bin "${GO_METALINTER_VERSION}"
+
+script:
+  # Lint text-like files
+  - scripts/pre-commit.sh --require-all
+  - make vendor-install
+  # Lint golang sources
+  - >
+    gometalinter -j4 --sort=path --sort=line --sort=column
+    --exclude="method NodeGetId should be NodeGetID"
+    --deadline 1h --vendor ./...
+  # Build the code & run tests
+  - make
+  - make test
+  # Build container w/ Docker
+  - docker version
+  - docker build -t gluster-csi-driver -f pkg/glusterfs/Dockerfile .
```

Comment:
don't we need to build container only if `Dockerfile` changes?

Reply:
The container depends on numerous things:
- The base image
- The packages installed by `yum`
- The location and existence of the built artifact

It's probably good to ensure all of these items continue to be valid.

BTW, the container built here is discarded after the CI job is done. Travis is not going to be used to upload.
#36 will enable Quay & Docker hub to do the entire build themselves upon commit to master or image tagging.